test/functional/wallet_descriptor.py

```diff
@@ -0,0 +1,137 @@
+#!/usr/bin/env python3
+# Copyright (c) 2019 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test descriptor wallet function."""
+
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import (
+    assert_equal,
+    assert_raises_rpc_error
+)
+
+
+class WalletDescriptorTest(BitcoinTestFramework):
+    def set_test_params(self):
+        self.setup_clean_chain = True
+        self.num_nodes = 1
+        self.extra_args = [['-keypool=100']]
+
+    def skip_test_if_missing_module(self):
+        self.skip_if_no_wallet()
+
+    def run_test(self):
+        # Make a descriptor wallet
+        self.log.info("Making a descriptor wallet")
+        self.nodes[0].createwallet(wallet_name="desc1", descriptors=True)
+        self.nodes[0].unloadwallet("")
+
+        # A descriptor wallet should have 100 addresses * 3 types = 300 keys
+        self.log.info("Checking wallet info")
+        wallet_info = self.nodes[0].getwalletinfo()
+        assert_equal(wallet_info['keypoolsize'], 300)
+        assert_equal(wallet_info['keypoolsize_hd_internal'], 300)
+
+        # Check that getnewaddress works
```

Comment:
need to check they're actually returning the right type as well for all 3 cases (maybe this is caught in a `--descriptor` variant of tests)

Reply:
Done